comms/dht/src/connectivity/metrics.rs

```diff
@@ -188,9 +210,9 @@ impl<T> TimeSeries<T> {
             "TimeSeries::values exceeded capacity"
         );
         if self.values.len() == self.values.capacity() {
```

Comment:
I just remembered that a `VecDeque`'s capacity cannot be relied upon. [playground example](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=c3e29115f7c466fedeb6def2894d1fc2)

I suggest keeping a separate `capacity: usize` field in `TimeSeries`